### PR TITLE
DISPATCH-1453 - Fix the code so the correct error message shows up wh…

### PR DESCRIPTION
**Pull request by ganeshmurthy**
…en a tx sender attaches in the absence of a coordinated link route

### CODE REVIEW

**Comment by ganeshmurthy** [comment]
@gemmellr Please take a look and approve if this works and you are satisfied. Thanks.

**Comment by gemmellr** [comment]
@ganeshmurthy this seems to do the trick, with links to the coordinator rejected with same specific error condition+description regardless whether "defaultDistribution : unavailable" is configured or not.

File: tests/system_tests_one_router.py [hunk 1]
```diff
@@ -3160,6 +3159,86 @@ def on_message(self, event):
...
+        self.assertTrue(test.passed)
+
+
+class RejectCoordinatorGoodMessageTest(RejectCoordinatorTest):
+    def __init__(self, url, link_route_present=False):
+        super(RejectCoordinatorGoodMessageTest, self).__init__(url)
+        self.link_route_present = link_route_present
+        self.error_with_link_route = "No route to the destination node"
+
+    def on_link_error(self, event):
```
Comment (kgiusti) [inline tests/system_tests_one_router.py]: Duplicate def of on_link_error (see next def below)

Comment (ganeshmurthy) [reply tests/system_tests_one_router.py]: Removed duplicate on_link_error. thanks

**Comment by ganeshmurthy** [comment]
@gemmellr please approve this PR when you get a chance.

**Comment by gemmellr** [comment]
The non-test code didnt seem to change since my previous comment, so I dont think theres anything needing me to try it out again is there? That being the case, still fine with me.

**Comment by ganeshmurthy** [comment]
> The non-test code didnt seem to change since my previous comment, so I dont think theres anything needing me to try it out again is there? That being the case, still fine with me.

That is indeed the case. No need to try it out again. I will squash and merge this PR to master. 
Thanks @gemmellr and @kgiusti

**Pull request closed without merging**